frontend/src/main/java/cz/cuni/mff/xrg/odcs/frontend/gui/views/PipelineEdit.java

```diff
@@ -968,12 +973,16 @@ private void setupButtons(boolean isModified) {
         setupButtons(isModified, this.pipeline.getId() == null);
     }
 
-    private void savePipelineAsNew() {
+    private boolean savePipelineAsNew() {
         if (!pipelineFacade.isUpToDate(pipeline)) {
         }
         if (!validate()) {
-            return;
+            return false;
         }
+        if (!pipelineCanvas.validateGraph()) {
```

Comment:
This test on validity of graph should be removed, it is too restrictive. It must work in a way that "If I click validate, I got issues". Nevertheless it must be possible to save pipeline which is not 100% valid (as it was working until now) Motivation: Sometimes, preparing pipeline is matter of hours, days, so I cannot be restricted by being able to store only valid pipelines. Also when debugging, the pipeline may not need to be 100% valid.